erdos/src/dataflow/graph/graph.rs

```diff
@@ -0,0 +1,342 @@
+use std::sync::{Arc, Mutex};
+
+use serde::Deserialize;
+
+use crate::{
+    dataflow::{
+        operator::{
+            OneInOneOut, OneInTwoOut, ParallelOneInOneOut, ParallelOneInTwoOut, ParallelSink,
+            ParallelTwoInOneOut, Sink, Source, TwoInOneOut,
+        },
+        stream::{ExtractStream, IngestStream, OperatorStream},
+        AppendableState, Data, LoopStream, State, Stream,
+    },
+    OperatorConfig,
+};
+
+use super::{InternalGraph, JobGraph};
+
+pub struct Graph {
+    internal_graph: Arc<Mutex<InternalGraph>>,
+}
+
+impl Default for Graph {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Graph {
+    pub fn new() -> Self {
+        Self {
+            internal_graph: Arc::new(Mutex::new(InternalGraph::new())),
+        }
+    }
+
+    pub fn get_ingest_stream<D>(&self, name: &str) -> IngestStream<D>
```

Comment:
I'm not a fan of these names. I'd prefer `add_ingest_stream()` or even `ingest_stream()`, since this function creates rather than retrieves. We may even want to go further with renaming, e.g. `add_ingress()` and `add_egress()` since this is more in-line with graph terminology, and leaves the door open to nested graphs.

@sukritkalra please let me know your thoughts as well!

Reply:
I would like to move it to `add_ingress`, but I'm also in favor of removing `*_extract_stream` methods from the `graph`, and just adding a LINQ style `extract` method on `OperatorStream`s instead. However, `add_ingress` and `extract` don't really seem to go well together.

Can we change this to `add_ingest_stream()` for now, and revisit this when we build a `SubGraph` abstraction? (which should hopefully be soon)

Reply:
What about `add_ingress` and `add_egress`? And then adding `to_egress()` on streams. We might also want to rename `IngestStream` to `IngressStream` and `ExtractStream` to `EgressStream`.

Reply:
I agree with this change. @objorkman could you make the corresponding changes please?